src/vivarium/framework/values/manager.py

```diff
@@ -153,17 +186,72 @@ def register_value_modifier(
             before the pipeline modifier is called.
         required_resources
             A list of resources that need to be properly sourced before the
-            pipeline modifier is called.  This is a list of strings, pipeline
+            pipeline modifier is called. This is a list of strings, pipeline
             names, or randomness streams.
         """
         pipeline = self.get_value(value_name)
-        value_modifier = pipeline.get_value_modifier(modifier, component)
-        self.logger.debug(f"Registering {value_modifier.name} as modifier to {value_name}")
+        self._configure_modifier(
+            pipeline,
+            modifier,
+            component,
+            requires_columns,
+            requires_values,
+            requires_streams,
+            required_resources,
+        )
 
-        dependencies = self._convert_dependencies(
-            modifier, requires_columns, requires_values, requires_streams, required_resources
+    def register_attribute_modifier(
+        self,
+        value_name: str,
+        modifier: Callable[..., Any],
+        # TODO [MIC-5452]: all calls should have a component
+        component: Component | Manager | None = None,
+        requires_columns: Iterable[str] = (),
+        requires_values: Iterable[str] = (),
+        requires_streams: Iterable[str] = (),
+        required_resources: Sequence[str | Resource] = (),
+    ) -> None:
+        """Marks a ``Callable`` as the modifier of a named attribute.
+
+        Parameters
+        ----------
+        value_name :
+            The name of the dynamic attribute pipeline to be modified.
+        modifier :
+            A function that modifies the source of the dynamic attribute pipeline
+            when called. If the pipeline has a ``replace_combiner``, the
+            modifier should accept the same arguments as the pipeline source
```

Comment:
I'm not sure if you are just being consistent across docstrings but I don't like "should accept" when it really HAS to have the same arguments.

Reply:
Updated "should accept" to "must have"